File: Data Structures & Algorithms/max-consecutive-ones/submission-2.ts
class Solution {
    /**
     * @param {number[]} nums
     * @return {number}
     */
    findMaxConsecutiveOnes(nums: number[]): number {
        let max = 0;
        let curr = 0;

        for(let i = 0; i < nums.length; i++) {
            if (nums[i] === 1) {
                curr++;
            } else {
                max = Math.max(curr, max);
                curr = 0;
            }
        }
        return Math.max(curr, max);
    }
}

File: Data Structures & Algorithms/replace-elements-with-greatest-element-on-right-side/submission-2.py
class Solution:
    def replaceElements(self, arr: List[int]) -> List[int]:
        rmax = -1
        R = len(arr) - 1

        while R >= 0:
            n = arr[R]
            arr[R] = rmax
            rmax = max(rmax, n)

            R -= 1

        return arr 
        